src/com/xilinx/rapidwright/interchange/EnumerateCellBelMapping.java

```diff
@@ -741,34 +741,42 @@ public static void populateCellBelPin(Enumerator<String> allStrings, Map<SiteTyp
             }
 
             for (List<String> parameters : getParametersFor(series, cell.getName())) {
+                HashSet<Map.Entry<String, String>> pinMapping = null;
+
                 for (Site site : siteMap.get(siteType)) {
-                    SiteInst siteInst = design.createSiteInst("test_site", siteType, site);
+                    if (pinMapping == null) {
+                        SiteInst siteInst = design.createSiteInst("test_site", siteType, site);
 
-                    String[] parameterArray = parameters.toArray(new String[parameters.size()]);
-                    physCell = design.createAndPlaceCell("test", Unisim.valueOf(cell.getName()), site.getName() + "/" + bel, parameterArray);
+                        String[] parameterArray = parameters.toArray(new String[parameters.size()]);
+                        physCell = design.createAndPlaceCell("test", Unisim.valueOf(cell.getName()),
+                                site.getName() + "/" + bel, parameterArray);
 
-                    // Build complete P2L map.
-                    Map<String, String> pinMap = new HashMap<String, String>();
+                        // Build complete P2L map.
+                        Map<String, String> pinMap = new HashMap<String, String>();
 
-                    for (Map.Entry<String, Set<String>> pinPair : physCell.getPinMappingsL2P().entrySet()) {
-                        for (String physPin : pinPair.getValue()) {
-                            pinMap.put(physPin, pinPair.getKey());
+                        for (Map.Entry<String, Set<String>> pinPair : physCell.getPinMappingsL2P().entrySet()) {
+                            for (String physPin : pinPair.getValue()) {
+                                pinMap.put(physPin, pinPair.getKey());
+                            }
                         }
-                    }
 
-                    pinMap.putAll(physCell.getPinMappingsP2L());
+                        pinMap.putAll(physCell.getPinMappingsP2L());
 
-                    HashSet<Map.Entry<String, String>> pinMapping = new HashSet<Map.Entry<String, String>>();
-                    for (Map.Entry<String, String> pinPair : pinMap.entrySet()) {
-                        pinMapping.add(new AbstractMap.SimpleEntry<String, String>(pinPair.getValue(), pinPair.getKey()));
-                    }
+                        pinMapping = new HashSet<Map.Entry<String, String>>();
+                        for (Map.Entry<String, String> pinPair : pinMap.entrySet()) {
+                            pinMapping.add(
```

Comment:
I'm not sure I understand what's going on here. `pinMapping` appears a `Set` of a pair of Strings -- is there need for it to be a set at all?

`pinMap` seems to be (unique) physical -> logical mapping created from logical -> physical mapping. Then `pinMapping` seems to reverse that again?

Reply:
The previous author of this code probably saw `Map.Entry<>` as a Java proxy for the Python tuple construct and used it judiciously.  This entire section of code should be re-written, but I believe that is beyond the scope of this PR.  The only change here that is relevant is that we are avoiding the re-computation of the map for every site when the map will be identical for all sites of the same type.

Reply:
Okay, let's make a note that this may be sped up further.

Reply:
See #659